elrond-wasm-node/src/api/send_api_node.rs

```diff
@@ -310,12 +328,34 @@ impl SendApi<ArwenBigUint> for ArwenApiImpl {
 		// account-level built-in function, so the destination address is the contract itself
 		let own_address = self.get_sc_address();
 
-		self.execute_on_dest_context(
+		let _ = self.execute_on_dest_context(
 			gas,
 			&own_address,
 			&ArwenBigUint::from(0u32),
 			function,
 			&arg_buffer,
-		)
+		);
+	}
+}
+
+/// Retrieves already pushed results, via `finish`.
+unsafe fn get_return_data(return_index: i32) -> BoxedBytes {
+	let len = getReturnDataSize(return_index);
+	let mut res = BoxedBytes::allocate(len as usize);
+	if len > 0 {
+		let _ = getReturnData(return_index, res.as_mut_ptr());
+	}
+	res
+}
+
+/// Retrieves already pushed results, via `finish`.
+unsafe fn get_return_data_range(from_index: i32, to_index: i32) -> Vec<BoxedBytes> {
+	let num_results = to_index - from_index;
+	let mut result = Vec::with_capacity(num_results as usize);
+	if num_results > 0 {
+		for index in from_index..to_index {
```

Comment:
Completely optional, but it might be more readable to use `from_index..=to_index` here, and call `get_return_data_range(num_return_data_before, num_return_data_after - 1)`. Variable name `to_index` suggests index-up-to-and-inclusively, but it is not used as such.

In case you make this change, `let num_results = to_index - from_index + 1` will be needed above.

Reply:
the convention "inclusive lower bound, exclusive upper bound" is quite pervasive, it is idiomatic of Rust and other languages, and in fact quite easy to work with. Even the simple `for` below has it. I am, however in favour of a better name

Reply:
or a comment, I think it might be enough here